In [70]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [71]:
#import data
content_df = r'C:\PROJECT\Accenture_internship\Content.csv'



# Define the file path

# Read the CSV file into a DataFrame
content_df = pd.read_csv( r'C:\PROJECT\Accenture_internship\Content.csv')



# Display the first few rows of the DataFrame to verify that it was loaded correctly
content_df.head()

     

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

In [72]:
content_df.info()
     

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  1000 non-null   int64 
 1   Content ID  1000 non-null   object
 2   User ID     1000 non-null   object
 3   Type        1000 non-null   object
 4   Category    1000 non-null   object
 5   URL         801 non-null    object
dtypes: int64(1), object(5)
memory usage: 47.0+ KB


In [73]:
#Summary of the data
content_df.describe()

Unnamed: 0
count  1000.000000
mean    499.500000
std     288.819436
min       0.000000
25%     249.750000
50%     499.500000
75%     749.250000
max     999.000000

In [74]:
content_df.drop(labels=['User ID','URL'],axis=1,inplace=True)
     
content_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  1000 non-null   int64 
 1   Content ID  1000 non-null   object
 2   Type        1000 non-null   object
 3   Category    1000 non-null   object
dtypes: int64(1), object(3)
memory usage: 31.4+ KB


In [75]:
#Check values in category columns
content_df['Category'].value_counts()
     

technology           71
animals              67
travel               67
culture              63
science              63
fitness              61
food                 61
healthy eating       61
cooking              60
soccer               58
tennis               58
education            57
dogs                 56
studying             55
veganism             48
public speaking      48
Fitness               5
Animals               4
Science               4
"soccer"              3
"culture"             3
Soccer                3
"dogs"                2
Education             2
Studying              2
Travel                2
Food                  2
"veganism"            1
"public speaking"     1
Public Speaking       1
"technology"          1
"cooking"             1
Healthy Eating        1
"studying"            1
"food"                1
Culture               1
"tennis"              1
Technology            1
"animals"             1
Veganism              1
"science"             1
Name: Category, 

In [76]:
#strip quote mark and lowercase all values
content_df['Category'] = content_df['Category'].apply(lambda x:x.strip('"').lower())
     

In [77]:
#checking if all categories are in lower case
content_df['Category'].value_counts()
     

technology         73
animals            72
travel             69
science            68
culture            67
fitness            66
food               64
soccer             64
healthy eating     62
cooking            61
tennis             59
education          59
studying           58
dogs               58
public speaking    50
veganism           50
Name: Category, dtype: int64

In [78]:
#rename content to content_type
content_df.rename(columns={'Type':'Content_Type'},inplace=True)
content_df['Content_Type'].value_counts()

photo    261
video    259
GIF      244
audio    236
Name: Content_Type, dtype: int64

In [79]:
#import 2nd dataset
reactions_df = pd.read_csv(r'C:\PROJECT\Accenture_internship\Reactions.csv',index_col=0)
reactions_df.head()

Content ID                               User ID  \
0  97522e57-d9ab-4bd6-97bf-c24d952602d2                                   NaN   
1  97522e57-d9ab-4bd6-97bf-c24d952602d2  5d454588-283d-459d-915d-c48a2cb4c27f   
2  97522e57-d9ab-4bd6-97bf-c24d952602d2  92b87fa5-f271-43e0-af66-84fac21052e6   
3  97522e57-d9ab-4bd6-97bf-c24d952602d2  163daa38-8b77-48c9-9af6-37a6c1447ac2   
4  97522e57-d9ab-4bd6-97bf-c24d952602d2  34e8add9-0206-47fd-a501-037b994650a2   

      Type             Datetime  
0      NaN  2021-04-22 15:17:15  
1  disgust  2020-11-07 09:43:50  
2  dislike  2021-06-17 12:22:51  
3   scared  2021-04-18 05:13:58  
4  disgust  2021-01-06 19:13:01

In [80]:
reactions_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25553 entries, 0 to 25552
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  25553 non-null  object
 1   User ID     22534 non-null  object
 2   Type        24573 non-null  object
 3   Datetime    25553 non-null  object
dtypes: object(4)
memory usage: 998.2+ KB


In [81]:
reactions_df.describe()

Content ID  \
count                                  25553   
unique                                   980   
top     4b2d0fff-3b4f-43ca-a7df-c430479cb9ba   
freq                                      49   

                                     User ID   Type             Datetime  
count                                  22534  24573                25553  
unique                                   500     16                25542  
top     c76c3393-88e2-47b0-ac37-dc4f2053f5a5  heart  2020-10-29 20:51:08  
freq                                      65   1622                    2

In [82]:
# drop user ID column
reactions_df.drop(labels = ['User ID'],axis=1,inplace=True)
     


In [83]:
# convert dates to datetime
reactions_df['Datetime'] = pd.to_datetime(reactions_df['Datetime'])


In [84]:
#drop rows with missing values
reactions_df.dropna(inplace=True)
     

In [85]:
#rename column
reactions_df.rename(columns={"Type":'Reaction_Type'},inplace=True)
     

In [86]:
reactions_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24573 entries, 1 to 25552
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Content ID     24573 non-null  object        
 1   Reaction_Type  24573 non-null  object        
 2   Datetime       24573 non-null  datetime64[ns]
dtypes: datetime64[ns](1), object(2)
memory usage: 767.9+ KB


In [87]:
#import 3rd Dataset
reactiontypes_df = pd.read_csv(r'C:\PROJECT\Accenture_internship\ReactionTypes.csv')
reactiontypes_df.head()
     

Unnamed: 0        Type Sentiment  Score
0           0       heart  positive     60
1           1        want  positive     70
2           2     disgust  negative      0
3           3        hate  negative      5
4           4  interested  positive     30

In [88]:
reactiontypes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  16 non-null     int64 
 1   Type        16 non-null     object
 2   Sentiment   16 non-null     object
 3   Score       16 non-null     int64 
dtypes: int64(2), object(2)
memory usage: 644.0+ bytes


In [89]:
reactiontypes_df.describe()

Unnamed: 0      Score
count   16.000000  16.000000
mean     7.500000  39.625000
std      4.760952  26.901983
min      0.000000   0.000000
25%      3.750000  14.250000
50%      7.500000  40.000000
75%     11.250000  66.250000
max     15.000000  75.000000

In [90]:
#rename type column
reactiontypes_df.rename(columns={'Type':'Reaction_Type'},inplace=True)
reactiontypes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     16 non-null     int64 
 1   Reaction_Type  16 non-null     object
 2   Sentiment      16 non-null     object
 3   Score          16 non-null     int64 
dtypes: int64(2), object(2)
memory usage: 644.0+ bytes


In [91]:
reactiontypes_df.value_counts()

Unnamed: 0  Reaction_Type  Sentiment  Score
0           heart          positive   60       1
1           want           positive   70       1
2           disgust        negative   0        1
3           hate           negative   5        1
4           interested     positive   30       1
5           indifferent    neutral    20       1
6           love           positive   65       1
7           super love     positive   75       1
8           cherish        positive   70       1
9           adore          positive   72       1
10          like           positive   50       1
11          dislike        negative   10       1
12          intrigued      positive   45       1
13          peeking        neutral    35       1
14          scared         negative   15       1
15          worried        negative   12       1
dtype: int64

# "Our data has been thoroughly cleansed in accordance with the client's requirements. This process involved removing any rows containing missing values, eliminating superfluous columns deemed irrelevant to the analysis, and ensuring consistency between data types and their respective column values."

# "Constructing a consolidated table to facilitate data modeling and analysis."(DATA MODELLING)

In [92]:
#merge tables on content ID columns
top_performers = reactions_df.merge(content_df,on='Content ID')
top_performers

Content ID Reaction_Type            Datetime  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust 2020-11-07 09:43:50   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2       dislike 2021-06-17 12:22:51   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2        scared 2021-04-18 05:13:58   
3      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust 2021-01-06 19:13:01   
4      97522e57-d9ab-4bd6-97bf-c24d952602d2    interested 2020-08-23 12:25:58   
...                                     ...           ...                 ...   
24568  75d6b589-7fae-4a6d-b0d0-752845150e56       dislike 2020-06-27 09:46:48   
24569  75d6b589-7fae-4a6d-b0d0-752845150e56     intrigued 2021-02-16 17:17:02   
24570  75d6b589-7fae-4a6d-b0d0-752845150e56    interested 2020-09-12 03:54:58   
24571  75d6b589-7fae-4a6d-b0d0-752845150e56       worried 2020-11-04 20:08:31   
24572  75d6b589-7fae-4a6d-b0d0-752845150e56       cherish 2021-01-04 04:55:11   

       Unnamed: 0 Content_Type    Category  
0               0        photo    studying  
1               0        photo    studying  
2               0        photo    studying  
3               0        photo    studying  
4               0        photo    studying  
...           ...          ...         ...  
24568         999        audio  technology  
24569         999        audio  technology  
24570         999        audio  technology  
24571         999        audio  technology  
24572         999        audio  technology  

[24573 rows x 6 columns]

In [93]:
#merge tables on reaction type column
top_performers = top_performers.merge(reactiontypes_df,on='Reaction_Type')
top_performers

Content ID Reaction_Type            Datetime  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust 2020-11-07 09:43:50   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust 2021-01-06 19:13:01   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust 2021-04-09 02:46:20   
3      9f737e0a-3cdd-4d29-9d24-753f4e3be810       disgust 2021-03-28 21:15:26   
4      230c4e4d-70c3-461d-b42c-ec09396efb3f       disgust 2020-08-04 05:40:33   
...                                     ...           ...                 ...   
24568  435007a5-6261-4d8b-b0a4-55fdc189754b         adore 2020-10-04 22:26:33   
24569  435007a5-6261-4d8b-b0a4-55fdc189754b         adore 2020-09-18 10:50:50   
24570  4e4c9690-c013-4ee7-9e66-943d8cbd27b7         adore 2020-10-31 03:58:44   
24571  4e4c9690-c013-4ee7-9e66-943d8cbd27b7         adore 2020-06-25 15:12:29   
24572  4e4c9690-c013-4ee7-9e66-943d8cbd27b7         adore 2020-12-17 16:32:57   

       Unnamed: 0_x Content_Type        Category  Unnamed: 0_y Sentiment  \
0                 0        photo        studying             2  negative   
1                 0        photo        studying             2  negative   
2                 0        photo        studying             2  negative   
3                 1        photo  healthy eating             2  negative   
4                 2        photo  healthy eating             2  negative   
...             ...          ...             ...           ...       ...   
24568           997        audio        veganism             9  positive   
24569           997        audio        veganism             9  positive   
24570           998          GIF         culture             9  positive   
24571           998          GIF         culture             9  positive   
24572           998          GIF         culture             9  positive   

       Score  
0          0  
1          0  
2          0  
3          0  
4          0  
...      ...  
24568     72  
24569     72  
24570     72  
24571     72  
24572     72  

[24573 rows x 9 columns]

# "Utilizing our dataset of 24,573 rows to identify the top five most popular categories."

# "Top Category Analysis: Exploring Categories Based on Popularity"

# "To determine category popularity, we sum the 'Score' column values by category."

In [94]:
top_five = top_performers.groupby('Category').sum().sort_values(by='Score',ascending=False)
top_five.head(5)

C:\Users\HP\AppData\Local\Temp\ipykernel_7128\2080812189.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  top_five = top_performers.groupby('Category').sum().sort_values(by='Score',ascending=False)


Unnamed: 0_x  Unnamed: 0_y  Score
Category                                         
animals              1027827         14298  74965
science              1019907         13315  71168
healthy eating        806218         12624  69339
technology            844401         12357  68738
food                  811558         12484  66676

# "Here are the top five categories by aggregate popularity: animals, science, healthy eating, technology, and food. We'll export this cleaned dataset to a CSV for sharing with the client."

In [95]:
top_performers.to_csv('category_data.csv')
     

In [97]:
file_path = r'C:\PROJECT\Accenture_internship\category_data.csv'
top_performers.to_csv(r'C:\PROJECT\Accenture_internship\category_data.csv', index=False)


PermissionError: [Errno 13] Permission denied: 'C:\\PROJECT\\Accenture_internship\\category_data.csv'